src/UiPath.Workflow/Debugger/ActivityExtensions.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Activities;
+using System.Activities.Runtime;
+using System.Activities.Tracking;
+using System.Activities.Validation;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace UiPath.Workflow.Debugger;
+
+public static class ActivityExtensions
+{
+    /// <summary>
+    /// A client may change properties at runtime (while execution is paused),
+    /// and so, based on the activity internal logic, a new call to CacheMetadata may be required.
+    /// This extension method will update the activity state.
+    /// ATTENTION - calling this method will generate a new activity.Id
+    /// </summary>
+    /// <returns>a list of validation errors</returns>
+    public static IList<ValidationError> RefreshActivityTree(this Activity activity)
```

Comment:
Please rename this to something more meaningful, like RefreshActivityArguments. This way we "isolate" more what we're doing here, in case we change it in the future.